contracts/BSCValidatorSet.sol

```diff
@@ -67,6 +67,11 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
   // key is the `consensusAddress` of `Validator`,
   mapping(address => MaintainInfo) public maintainInfoMap;
 
+  // BEP-131 candidate validator
+  uint256 public numOfCabinets = 21;
```

Comment:
When we do upgrade, we can only upgrade the code of smart contract, so the storage of this contract will not change, it means `numOfCabinets` will be zero after we upgrade it. So you have to assume `numOfCabinets`, `maxNumOfCandidates` and `maxNumOfWorkingCandidates` is zero before we change the value of them through governance .